pages/docs/[[...slug]].tsx

```diff
@@ -1,82 +1,30 @@
-import { InferGetStaticPropsType } from 'next';
-import { MDXRemote } from 'next-mdx-remote';
-import shell from 'shelljs';
-import fs from 'fs';
-import locales from '../../i18n/locales';
+import { allDocs } from '.contentlayer/data';
+import type { Doc } from '.contentlayer/types';
 import { MDXComponents } from 'components/mdx-components';
+import { GetStaticPaths, GetStaticProps } from 'next';
+import { useMDXComponent } from 'next-contentlayer/hooks';
 import Layout from 'layouts';
-import path from 'path';
-import { loadMdx } from 'utils/load-mdx';
 
-const CONTENT_PATH = 'docs';
-
-export default function Page({
-  mdxSource,
-  frontMatter,
-}: InferGetStaticPropsType<typeof getStaticProps>) {
+export default function Page({ doc }: { doc: Doc }) {
+  const Component = useMDXComponent(doc.body.code);
```

Comment:
Why do we use `useMDXComponent` here, but we memoize the result of `getMDXComponent` in `changelog.tsx`?

Reply:
This is a custom hook from the `contentlayer` library. It returns the compiled React code with the ability to swap rendered components. For the changelog. It's a different story since we're fetching the changelog manually and it doesn't go through `contentlayer`

Reply:
Ah okay! Good to know. Thanks @segunadebayo!